### PR TITLE
[Platform] Move model arch check to platform

### PR DESCRIPTION
**Pull request by MengqingCao**
Part of #11162

### CODE REVIEW

File: vllm/platforms/rocm.py [hunk 1]
```diff
@@ -102,6 +127,18 @@ def check_and_update_config(cls, vllm_config: VllmConfig) -> None:
             else:
                 parallel_config.worker_cls = "vllm.worker.worker.Worker"
 
+    @classmethod
+    def verify_model_arch(cls, model_arch: str) -> None:
```
Comment (DarkLight1337) [inline vllm/platforms/rocm.py:131]: Shouldn't it be `check_model_arch` here?

Comment (DarkLight1337) [reply vllm/platforms/rocm.py:131]: (Though personally I prefer `verify_model_arch` as the method name.

**Comment by MengqingCao** [comment]
Oh, thanks for pointing that. I previously named it verify_madel_arch, but after changing the name, I forgot to modify it here.

Then I will unify this method as verify_madel_arch in the next commit.



---- Replied Message ----
| From | Cyrus ***@***.***> |
| Date | 12/26/2024 21:01 |
| To | ***@***.***> |
| Cc | Mengqing ***@***.***>***@***.***> |
| Subject | Re: [vllm-project/vllm] [Platform] Move model arch check to platform (PR #11503) |

@DarkLight1337 commented on this pull request.

In vllm/platforms/rocm.py:

> @@ -102,6 +127,18 @@ def check_and_update_config(cls, vllm_config: VllmConfig) -> None:
             else:
                 parallel_config.worker_cls = "vllm.worker.worker.Worker"
 
+    @classmethod
+    def verify_model_arch(cls, model_arch: str) -> None:


(Though personally I prefer verify_model_arch as the method name.)

File: vllm/platforms/rocm.py (hunk 1 above)
Comment (MengqingCao) [reply vllm/platforms/rocm.py:131]: Thx! Done.

**Review by DarkLight1337 — APPROVED** [review]

**Pull request merged**